src/main/java/com/graphaware/module/uuid/generator/SequenceIdGenerator.java

```diff
@@ -0,0 +1,34 @@
+package com.graphaware.module.uuid.generator;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Result;
+
+import com.graphaware.common.uuid.UuidGenerator;
+import com.graphaware.module.GraphDatabaseServiceAware;
+
+/**
+ * A complex implementation of UuidGenerator that makes use of the provided GraphDatabaseService to 
+ * generate a numerically increasing sequence through Cypher.
+ */
+public class SequenceIdGenerator implements UuidGenerator, GraphDatabaseServiceAware {
+
+	public static final String cypher = "MERGE (sequenceMetadata:SequenceMetadata) set sequenceMetadata.sequence = coalesce(sequenceMetadata.sequence, 0) + 1 return sequenceMetadata.sequence";
```

Comment:
The `0` and `1` in the query can be parameterized, it doesn't change anything to how the query will behave but matters a lot for performance. By using parameters the query plan will be cached and re-used, also by the simplicity of the query in 3.0.6 and definitely in 3.1 chances are high that by using params this query will be compiled to java bytecode.